File: lib/features/about/about_screen.dart
import 'package:flutter/material.dart';
import '../../core/theme/app_spacing.dart';
import '../../core/widgets/bilingual_title.dart';

class AboutScreen extends StatelessWidget {
  const AboutScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Scaffold(
      appBar: AppBar(title: const BilingualTitle(telugu: 'గురించి & గోప్యత', english: 'About & privacy')),
      body: ListView(
        padding: const EdgeInsets.all(AppSpacing.lg),
        children: [
          Text('Telugu Panchangam & Jatakam', style: theme.textTheme.headlineSmall),
          const SizedBox(height: 4),
          Text('Version 1.0.0 (development preview)', style: theme.textTheme.bodySmall?.copyWith(color: theme.colorScheme.onSurfaceVariant)),
          const SizedBox(height: AppSpacing.xxl),
          _Card(
            title: 'About this build',
            body: 'This is a frontend-only development preview. Festival dates from 19 Mar 2026 through '
                '24 Dec 2026 are taken from a real published almanac (see Sources & references); daily '
                'Tithi/Nakshatra/Yoga/Karana values and Jatakam charts still come from a placeholder mock data '
                'layer pending a verified calculation engine. Values marked "Preview calculation" or '
                '"[VERIFIED CONTENT REQUIRED]" are not to be relied on for religious observance.',
          ),
          const SizedBox(height: AppSpacing.lg),
          _Card(
            title: 'Privacy',
            body: 'Your name, birth details and preferences are stored locally on this device only. Nothing is '
                'sent to a server — this app does not yet connect to any backend.',
          ),
          const SizedBox(height: AppSpacing.lg),
          _Card(
            title: 'Content approach',
            body: 'We treat mantras, festival rules and Jyotisha interpretation as content requiring citation from '
                'a qualified, authoritative source before it is shown as fact. See Sources & references for the '
                'current status of each citation.',
          ),
        ],
      ),
    );
  }
}

class _Card extends StatelessWidget {
  const _Card({required this.title, required this.body});
  final String title;
  final String body;

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Container(
      width: double.infinity,
      padding: const EdgeInsets.all(AppSpacing.lg),
      decoration: BoxDecoration(
        color: theme.colorScheme.surface,
        borderRadius: BorderRadius.circular(AppRadii.lg),
        border: Border.all(color: theme.colorScheme.outlineVariant),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(title, style: theme.textTheme.titleMedium),
          const SizedBox(height: AppSpacing.sm),
          Text(body, style: theme.textTheme.bodyMedium),
        ],
      ),
    );
  }
}

File: app/src/main/kotlin/com/panchangam/telugupanchangam/ui/about/AboutScreen.kt
package com.panchangam.telugupanchangam.ui.about

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.panchangam.telugupanchangam.data.FestivalData
import com.panchangam.telugupanchangam.domain.SourceReference
import com.panchangam.telugupanchangam.ui.components.OutlinedBlock
import com.panchangam.telugupanchangam.ui.components.SourceReferenceCard
import com.panchangam.telugupanchangam.ui.theme.Ink

@Composable
fun AboutScreen() {
    val sources: List<SourceReference> = remember {
        listOf(FestivalData.drikPanchang2026, FestivalData.pendingAlmanac, FestivalData.pendingMantra)
    }

    LazyColumn(
        modifier = Modifier.fillMaxSize().background(Ink.White),
        contentPadding = PaddingValues(start = 16.dp, end = 16.dp, top = 16.dp, bottom = 32.dp)
    ) {
        item {
            Text("గురించి", style = MaterialTheme.typography.headlineMedium)
            Text("About & sources", style = MaterialTheme.typography.bodySmall)
            Spacer(Modifier.height(4.dp))
            Text("Version 1.0.0 (development preview)", style = MaterialTheme.typography.bodySmall)
            Spacer(Modifier.height(20.dp))
        }

        item {
            InfoCard(
                "About this build",
                "Festival dates from 19 Mar 2026 through 24 Dec 2026 are taken from a real published almanac " +
                    "(cited below). Daily Tithi, Nakshatra, Yoga, Karana, sunrise/sunset and Choghadiya values " +
                    "are produced by a development placeholder, not a verified astronomical engine — anything " +
                    "labelled \"Preview calculation\" should not be relied on for religious observance."
            )
        }

        item {
            InfoCard(
                "How we handle provenance",
                "Every festival date, rule and mantra in this app is meant to carry a source — title, edition, " +
                    "language and a verification status. Where a qualified citation has not been reviewed, the " +
                    "content is withheld and marked [VERIFIED CONTENT REQUIRED] rather than guessed at."
            )
        }

        item {
            InfoCard(
                "Privacy",
                "This app stores nothing about you and makes no network requests. There is no account, no " +
                    "analytics and no backend."
            )
        }

        item {
            Spacer(Modifier.height(24.dp))
            Text("ఆధారాలు  ·  Sources", style = MaterialTheme.typography.headlineSmall)
            Spacer(Modifier.height(8.dp))
        }
        items(sources) { source ->
            SourceReferenceCard(source, Modifier.padding(bottom = 8.dp))
        }
    }
}

@Composable
private fun InfoCard(title: String, body: String) {
    Column(Modifier.padding(bottom = 12.dp)) {
        OutlinedBlock {
            Text(title, style = MaterialTheme.typography.titleMedium)
            Spacer(Modifier.height(6.dp))
            Text(body, style = MaterialTheme.typography.bodyMedium)
        }
    }
}
